src/MEDS_transforms/extract/split_and_shard_subjects.py

```diff
@@ -93,6 +93,8 @@ def shard_subjects(
         >>> shard_subjects(subjects, 3, external_splits)
         {'train/0': [5, 1, 3], 'train/1': [2, 6, 4], 'test/0': [10, 7], 'test/1': [8, 9]}
     """
+    if split_fracs_dict is None:
```

Comment:
With this check you can update the if statement on line 119 as well, as it can no longer be `None`

Reply:
I also don't think this is semantically equivalent; the doctest on line 91 seems to expect different behavior.